app/controllers/microposts_controller.rb

```diff
@@ -1,34 +1,32 @@
 class MicropostsController < ApplicationController
-  before_action :user_signed_in?, only: [:create, :destroy]
+  before_action :user_signed_in?, only: %i[create destroy]
   before_action :correct_user, only: [:destroy]
 
   def create
-    @micropost = current_user.microposts.build(micropost_params)
-    if @micropost.save
-      flash[:success] = "Micropost created."
+    @micropost = DelayedMicropostForm.new(micropost_params)
+    if @micropost.submit
+      flash[:success] = 'Micropost created.'
       redirect_to root_url
-    else 
+    else
       render 'static_pages/home'
     end
   end
 
   def destroy
-    if @micropost.present?
-      authorize @micropost.destroy!
-    end
+    authorize @micropost.destroy! if @micropost.present?
```

Comment:
Need to authorize `@micropost` first. Here you destroy micropost first, and then check's deleted object policy. Also, you set flash message after redirect. I think flash message never set's because of that.

```
raise ActiveRecord::RecordNotFound unless @micropost

authorize @micropost
@micropost.destroy!
flash[:success] = 'Micropost deleted.'
redirect_to root_url
 
```

Reply:
thank you! Here the update https://github.com/saparjohnick/twitter_like/pull/1/commits/a7f2d7d827b7b638449b1488309662368d2bde42